sqlglot/dialects/dialect.py

```diff
@@ -655,6 +659,12 @@ def tokenizer(self) -> Tokenizer:
             self._tokenizer = self.tokenizer_class(dialect=self)
         return self._tokenizer
 
+    @property
+    def jsonpath_tokenizer(self) -> JSONPathTokenizer:
+        if not hasattr(self, "_jsonpath_tokenizer"):
+            self._jsonpath_tokenizer = self.jsonpath_tokenizer_class(dialect=self)
+        return self.jsonpath_tokenizer_class(dialect=self)
```

Comment:
This doesn't cache it right?
```suggestion
        return self._jsonpath_tokenizer
```

Reply:
Oh right, my intention was for `JSONPathTokenizer` to follow `Tokenizer` i.e this shouldn't create a new instance here. Will fix